File: src/main/java/br/com/repository/VotoRepository.java
package br.com.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import br.com.entity.Associado;
import br.com.entity.Pauta;
import br.com.entity.Voto;

@Repository
public interface VotoRepository extends JpaRepository<Voto, Long> {

	Optional<Voto> findByPautaAndAssociado(Pauta pauta, Associado associado);

	List<Voto> findByPauta(Pauta pauta);

}

File: src/main/kotlin/br/com/southsystem/desafiobackvotos/gateway/database/VotoRepository.kt
package br.com.southsystem.desafiobackvotos.gateway.database

import br.com.southsystem.desafiobackvotos.domain.Voto
import org.springframework.data.repository.reactive.ReactiveCrudRepository

interface VotoRepository: ReactiveCrudRepository<Voto, Long> {
}


